### PR TITLE
[Doc] Add `maxSubscriptionsPerTopic` to broker and standalone configurations

### PR DESCRIPTION
**Pull request by momo-jun**
### Modifications

To align doc with code https://github.com/apache/pulsar/pull/8289, the following changes have been made:
1. Add `maxSubscriptionsPerTopic` to both broker and standalone sections.
2. Copy/paste `maxProducersPerTopic`, `maxConsumersPerTopic` and `maxConsumersPerSubscription` from standalone configurations to broker configurations because these three parameters are in the broker.conf file.
3. Remove a duplicate parameter `maxNumPartitionsPerPartitionedTopic` from the standalone section.

> Please kindly note that the same changes have been applied to more historical versions since 2.7.0.

### Documentation

- [x] `doc`

### CODE REVIEW

**Comment by momo-jun** [comment]
Hi @hangc0276 @Anonymitaet can you please review this doc PR? Thanks.
Please kindly note that the same changes will be applied to more historical versions as long as it gets approval.

File: site2/docs/reference-configuration.md [hunk 1]
```diff
@@ -208,6 +208,10 @@ brokerServiceCompactionThresholdInBytes|If the estimated backlog size is greater
 |statusFilePath|  Path for the file used to determine the rotation status for the broker when responding to service discovery health checks ||
 |preferLaterVersions| If true, (and ModularLoadManagerImpl is being used), the load manager will attempt to use only brokers running the latest software version (to minimize impact to bundles)  |false|
 |maxNumPartitionsPerPartitionedTopic|Max number of partitions per partitioned topic. Use 0 or negative number to disable the check|0|
+| maxSubscriptionsPerTopic | Maximum number of subscriptions allowed to subscribe to a topic. Once this limit reaches, the broker rejects new subscriptions until the number of subscriptions decreases. When the value is set to 0, maxSubscriptionsPerTopic-limit check is disabled. | 0 |
```
Comment (Anonymitaet) [inline site2/docs/reference-configuration.md:211]:
```suggestion
| maxSubscriptionsPerTopic | Maximum number of subscriptions allowed to subscribe to a topic. Once this limit reaches, the broker rejects new subscriptions until the number of subscriptions decreases. When the value is set to 0, `maxSubscriptionsPerTopic-limit` check is disabled. | 0 |
```

File: site2/docs/reference-configuration.md [hunk 2]
```diff
@@ -208,6 +208,10 @@ brokerServiceCompactionThresholdInBytes|If the estimated backlog size is greater
...
+| maxProducersPerTopic | Maximum number of producers allowed to connect to a topic. Once this limit reaches, the broker rejects new producers until the number of connected producers decreases. When the value is set to 0, maxProducersPerTopic-limit check is disabled. | 0 |
```
Comment (Anonymitaet) [inline site2/docs/reference-configuration.md:212]:
```suggestion
| maxProducersPerTopic | Maximum number of producers allowed to connect to a topic. Once this limit reaches, the broker rejects new producers until the number of connected producers decreases. When the value is set to 0, `maxProducersPerTopic-limit` check is disabled. | 0 |
```

File: site2/docs/reference-configuration.md [hunk 3]
```diff
@@ -208,6 +208,10 @@ brokerServiceCompactionThresholdInBytes|If the estimated backlog size is greater
...
+| maxConsumersPerTopic | Maximum number of consumers allowed to connect to a topic. Once this limit reaches, the broker rejects new consumers until the number of connected consumers decreases. When the value is set to 0, maxConsumersPerTopic-limit check is disabled. | 0 |
```
Comment (Anonymitaet) [inline site2/docs/reference-configuration.md:213]:
```suggestion
| maxConsumersPerTopic | Maximum number of consumers allowed to connect to a topic. Once this limit reaches, the broker rejects new consumers until the number of connected consumers decreases. When the value is set to 0, `maxConsumersPerTopic-limit check` is disabled. | 0 |
```

File: site2/docs/reference-configuration.md [hunk 4]
```diff
@@ -208,6 +208,10 @@ brokerServiceCompactionThresholdInBytes|If the estimated backlog size is greater
...
+| maxConsumersPerSubscription | Maximum number of consumers allowed to connect to a subscription. Once this limit reaches, the broker rejects new consumers until the number of connected consumers decreases. When the value is set to 0, maxConsumersPerSubscription-limit check is disabled. | 0 |
```
Comment (Anonymitaet) [inline site2/docs/reference-configuration.md:214]:
```suggestion
| maxConsumersPerSubscription | Maximum number of consumers allowed to connect to a subscription. Once this limit reaches, the broker rejects new consumers until the number of connected consumers decreases. When the value is set to 0, `maxConsumersPerSubscription-limit` check is disabled. | 0 |
```

**Review by Anonymitaet — DISMISSED** [review]
Approve from tech writing pespective

**Review by RobertIndie — DISMISSED** [review]

**Review by Anonymitaet — APPROVED** [review]

**Comment by hangc0276** [comment]
Looks good to me.

**Pull request merged**